stripe-ui-core/src/main/java/com/stripe/android/uicore/utils/StateFlows.kt

```diff
@@ -96,6 +128,23 @@ fun <T1, T2, T3, R> combineAsStateFlow(
  * Combines three [StateFlow]s into another, instead of loosening the result to a [Flow].
  */
 @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+fun <T1, T2, T3, T4, R> combineAsStateFlow(
+    flow1: StateFlow<T1>,
+    flow2: StateFlow<T2>,
+    flow3: StateFlow<T3>,
+    flow4: StateFlow<T4>,
+    transform: (T1, T2, T3, T4) -> R,
+): StateFlow<R> {
+    return FlowToStateFlow(
+        flow = combine(flow1, flow2, flow3, flow4, transform).distinctUntilChanged(),
```

Comment:
I don't think the `distinctUntilChanged` is needed here, but I realize it's also in other combines. But it shouldn't be needed in those either. I'm fine cleaning these up separately though.